llvm/lib/SYCLLowerIR/TargetHelpers.cpp

```diff
@@ -56,6 +56,7 @@ void populateKernels(Module &M, SmallVectorImpl<KernelPayload> &Kernels,
   if (!AnnotationMetadata)
     return;
 
+  std::vector<MDNode *> PossibleDependencies;
```

Comment:
nitpicking: [LLVM Programmer's Manual](https://llvm.org/docs/ProgrammersManual.html#sequential-containers-std-vector-std-list-etc) says that you should not pick `std::vector`, unless you have a reason for that:

> Pick the first in this section that will do what you want.

And `SmallVector` is listed before `std::vector`